bindings/ergo-lib-c/src/lib.rs

```diff
@@ -70,11 +76,127 @@ pub unsafe extern "C" fn ergo_wallet_address_from_testnet(
     Error::c_api_from(res)
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn ergo_wallet_address_from_mainnet(
+    address_str: *const c_char,
+    address_out: *mut AddressPtr,
+) -> ErrorPtr {
+    let address = CStr::from_ptr(address_str).to_string_lossy();
+    let res = address_from_mainnet(&address, address_out);
+    Error::c_api_from(res)
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn ergo_wallet_address_from_base58(
+    address_str: *const c_char,
+    address_out: *mut AddressPtr,
+) -> ErrorPtr {
+    let address = CStr::from_ptr(address_str).to_string_lossy();
+    let res = address_from_base58(&address, address_out);
+    Error::c_api_from(res)
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn ergo_wallet_address_to_base58(
+    address: ConstAddressPtr,
+    network_prefix: NetworkPrefix,
+    _address_str: *mut *const c_char,
+) -> ErrorPtr {
+    let res = match address_to_base58(address, network_prefix) {
+        Ok(s) => {
+            *_address_str = CString::new(s).unwrap().into_raw();
+            Ok(())
+        }
+        Err(e) => Err(e),
+    };
+    Error::c_api_from(res)
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn ergo_wallet_address_type_prefix(
+    address: ConstAddressPtr,
+) -> ReturnNum<u8> {
+    match address_type_prefix(address) {
+        Ok(value) => ReturnNum {
+            value: value as u8,
+            error: std::ptr::null_mut(),
+        },
+        Err(e) => ReturnNum {
+            value: 0, // Just a dummy value
+            error: Error::c_api_from(Err(e)),
+        },
+    }
+}
+
 #[no_mangle]
 pub extern "C" fn ergo_wallet_address_delete(address: AddressPtr) {
     address_delete(address)
 }
 
+// -------------------------------------------------------------------------------------------------
+// BlockHeader functions ---------------------------------------------------------------------------
```

Comment:
Is it possible to use modules here? So that we don't end up with one huge lib.rs file.

Reply:
Yes for sure. I'll break it up.